### PR TITLE
Don't close API streams when one data channel closes

### PR DESCRIPTION
**Pull request by shibe-dev**
## Motivation
fix for issue #2075 

## Changes
improved the channel closing logic (setting a channel to nil stops it from being selected). This design would be an improvement over the current one but I'm unsure if this is sufficient/final (or if its even what you're looking for) - would love feedback @lrettig. Will leave as draft PR

### CODE REVIEW

File: api/grpcserver/globalstate_service.go [hunk 1]
```diff
@@ -230,16 +230,14 @@ func (s GlobalStateService) AccountDataStream(in *pb.AccountDataStreamRequest, s
...
-				// we could handle this more gracefully, by no longer listening
-				// to this stream but continuing to listen to the other stream,
-				// but in practice one should never be closed while the other is
-				// still running, so it doesn't matter
-				log.Info("account channel closed, shutting down")
-				return nil
+		case updatedAccount, accountOk := <-channelAccount:
+			if !accountOk {
+				log.Info("account channel closed")
+				channelAccount = nil
```
Comment (lrettig) [inline api/grpcserver/globalstate_service.go:240]: if `accountOk` is false, does that mean `updatedAccount` is nil? The next line reads `updatedAccount`, will this work okay if the channel has just been closed or do you need to add a check for this case? (here and in the others, below)

File: api/grpcserver/globalstate_service.go [hunk 2]
```diff
@@ -317,6 +307,8 @@ func (s GlobalStateService) AccountDataStream(in *pb.AccountDataStreamRequest, s
 		// that the service needs to shut down?
 		// See https://github.com/spacemeshos/go-spacemesh/issues/2075
 	}
+
+	return nil
```
Comment (lrettig) [inline api/grpcserver/globalstate_service.go:311]: Please add a comment here explaining what this line does and why/when it's necessary, the context is lost here

**Review by lrettig — COMMENTED** [review]
See https://github.com/spacemeshos/go-spacemesh/issues/2075#issuecomment-742014594 - I don't think this task is well-defined yet

File: api/grpcserver/globalstate_service.go (hunk 1 above)
Comment (shibe-dev) [reply api/grpcserver/globalstate_service.go:240]: im still rather new to golang but it seems like [setting a channel a to nil stops it from being selected](https://golangbyexample.com/select-with-nil-channel-golang/) which seems to be the logic you were suggesting that the API should go for

Comment (lrettig) [reply api/grpcserver/globalstate_service.go:240]: Yes, that may be true for the _next_ loop iteration, but what about the current iteration?

**Comment by antonlerner** [comment]
closed due to inactivity

**Pull request closed without merging**